airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -694,70 +715,35 @@ public static boolean isTerminal(final Pod pod) {
    * This method hits the Kube Api server to retrieve statuses. Most of the complexity here is
    * minimising the api calls for performance.
    */
-  private int getReturnCode(final Pod pod) {
-    if (returnCode != null) {
-      return returnCode;
-    }
-
-    final String podName = pod.getMetadata().getName();
-    final String podNamespace = pod.getMetadata().getNamespace();
-
-    // If there is no return code (see above) and we have checked the status recently, assume the pod is
-    // still running. We do this to avoid overloading the Kube Api server.
-    if (lastStatusCheck != null && System.currentTimeMillis() - lastStatusCheck < statusCheckInterval.toMillis()) {
-      throw new IllegalThreadStateException(prependPodInfo("Kube pod process has not exited yet. (cached)", podNamespace, podName));
-    }
-
-    final Pod refreshedPod = fabricClient.pods().inNamespace(podNamespace).withName(podName).get();
-    if (refreshedPod == null) {
-      if (wasKilled.get()) {
-        LOGGER.info(prependPodInfo(
-            String.format("Unable to find pod to retrieve exit value. Defaulting to  value %s. This is expected if the job was cancelled.",
-                KILLED_EXIT_CODE),
-            podNamespace, podName));
-        return KILLED_EXIT_CODE;
+  private int getReturnCode() {
+    if (exitCodeFuture.isDone()) {
```

Comment:
nit: this can be rewritten to avoid the if else statement and simplify the logic here. 

```
  if (!exitCodeFuture.isDone()) {
      throw new IllegalThreadStateException(prependPodInfo("Main container in kube pod has not terminated yet.",
          podDefinition.getMetadata().getNamespace(),
          podDefinition.getMetadata().getName()));
  }

  try {
      return exitCodeFuture.get();
  } catch (InterruptedException | ExecutionException e) {
    throw new RuntimeException(
        prependPodInfo("Cannot find pod %s : %s while trying to retrieve exit code. This probably means the pod was not correctly created.",
            podDefinition.getMetadata().getNamespace(),
            podDefinition.getMetadata().getName()));
  } 
```

Reply:
Personally I dislike splitting up a `if { } else { }` sections to include interrupting blocks for one of the statements.